pkg/pgmodel/migrations/sql/1_base_schema.up.sql

```diff
@@ -701,21 +702,25 @@ CREATE OR REPLACE  FUNCTION SCHEMA_PROM.series_id(label jsonb)
 RETURNS BIGINT AS $$
 DECLARE
   series_id bigint;
+  table_name name;
+  metric_id int;
 BEGIN
    --need to make sure the series partition exists
-   PERFORM SCHEMA_CATALOG.get_or_create_metric_table_name(label->>'__name__');
-
-   WITH CTE AS (
-       SELECT SCHEMA_PROM.label_array(label)
-   )
-   SELECT id
-   FROM SCHEMA_CATALOG.series
-   WHERE labels = (SELECT * FROM cte)
-         AND metric_id =  (SELECT (SCHEMA_CATALOG.get_metric_table_name_if_exists(label->>'__name__')).id)
-   UNION ALL
-   SELECT SCHEMA_CATALOG.create_series(id, table_name, (SELECT * FROM cte))
-   FROM SCHEMA_CATALOG.get_or_create_metric_table_name(label->>'__name__')
-   LIMIT 1
+   SELECT mtn.id, mtn.table_name FROM SCHEMA_CATALOG.get_or_create_metric_table_name(label->>'__name__') mtn
+   INTO metric_id, table_name;
+
+   EXECUTE format($query$
+    WITH CTE AS (
+        SELECT SCHEMA_PROM.label_array($1)
+    )
+    SELECT id
+    FROM SCHEMA_DATA_SERIES.%1$I as series
+    WHERE labels = (SELECT * FROM cte)
+    UNION ALL
+    SELECT SCHEMA_CATALOG.create_series(%2$L, %1$L, (SELECT * FROM cte))
+    LIMIT 1
+   $query$, table_name, metric_id)
+   USING label
```

Comment:
I still think this should be renamed to get or create and then we should have a get only one or something like that that is exposed. Especially given that it takes an exclusive lock on the table.